pages/api/Schema/CoachingAnswerSets/dataHandler.ts

```diff
@@ -0,0 +1,171 @@
+import {
+  CoachingAnswerSet,
+  CoachingAnswer,
+  CoachingQuestion,
+} from '../../../../graphql/types.generated';
+
+export type CoachingAnswerSetData = CoachingAnswerSetResponseData[];
+
+export type CoachingAnswerSetIncluded = (
+  | CoachingAnswerResponse
+  | CoachingQuestionResponse
+)[];
+
+interface CoachingAnswerSetResponseData {
+  id: string;
+  type: 'coaching_answer_sets';
+  attributes: {
+    completed_at: string;
+    created_at: string;
+    updated_at: string;
+  };
+  relationships: {
+    answers: {
+      data: {
+        id: string;
+        type: 'coaching_answers';
+      }[];
+    };
+    questions: {
+      data: {
+        id: string;
+        type: 'coaching_questions';
+      }[];
+    };
+  };
+}
+
+interface CoachingAnswerResponse {
+  id: string;
+  type: 'coaching_answers';
+  attributes: {
+    created_at: string;
+    response: string;
+    updated_at: string;
+  };
+}
+
+interface CoachingQuestionResponse {
+  id: string;
+  type: 'coaching_questions';
+  attributes: {
+    created_at: string;
+    position: number;
+    prompt: string;
+    required: boolean;
+    response_options: string[] | null;
+    updated_at: string;
+  };
+}
+
+const getCoachingAnswerSets = (
+  data: CoachingAnswerSetData,
+  included: CoachingAnswerSetIncluded,
+): CoachingAnswerSet[] => {
+  const response: CoachingAnswerSet[] = [];
+
+  data.forEach((answerSetData) => {
+    const {
+      id,
+      attributes: { completed_at, created_at, updated_at },
+    } = answerSetData;
+
+    //create answers
+    const answers = createCoachingAnswersList(answerSetData, included);
+    //create questions
+    const questions = createCoachingQuestionsList(answerSetData, included);
+
+    const answerSet: CoachingAnswerSet = {
+      id,
+      answers,
+      completedAt: completed_at,
+      createdAt: created_at,
+      questions,
+      updatedAt: updated_at,
+    };
+
+    response.push(answerSet);
+  });
+
+  return response;
+};
+
+const createCoachingAnswersList = (
+  answerSetData: CoachingAnswerSetResponseData,
+  includedItems: CoachingAnswerSetIncluded,
+): CoachingAnswer[] => {
+  const answers: CoachingAnswer[] = [];
+
+  const ids = answerSetData.relationships.answers.data.map(
+    (answer) => answer.id,
+  );
+
+  ids.forEach((id) => {
+    const answerItem = includedItems.find(
+      (item) => item.id === id,
```

Comment:
I think you want the type condition in here also. There's a chance (very improbably) that you find an id that is from another type and stop looking for the one you really want too soon.

Same in your question function below.